jvm-packages/xgboost4j-spark/src/test/scala/ml/dmlc/xgboost4j/scala/spark/XGBoostRegressorSuite.scala

```diff
@@ -93,6 +95,36 @@ class XGBoostRegressorSuite extends FunSuite with PerTest {
     }
   }
 
+  test("XGBoost-Spark XGBoostRegressor CrossValidator output should result in an estimator") {
```

Comment:
any specific type of error you want to prevent here?

Reply:
I've been seeing

    org.apache.spark.SparkException: Exception thrown in awaitResult:
    [info]   at org.apache.spark.util.ThreadUtils$.awaitResult(ThreadUtils.scala:205)
    [info]   at org.apache.spark.ml.tuning.CrossValidator$$anonfun$4$$anonfun$6.apply(CrossValidator.scala:164)
    [info]   at org.apache.spark.ml.tuning.CrossValidator$$anonfun$4$$anonfun$6.apply(CrossValidator.scala:164)
    [info]   at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
    [info]   at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
    [info]   at scala.collection.IndexedSeqOptimized$class.foreach(IndexedSeqOptimized.scala:33)
    [info]   at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:186)
    [info]   at scala.collection.TraversableLike$class.map(TraversableLike.scala:234)
    [info]   at scala.collection.mutable.ArrayOps$ofRef.map(ArrayOps.scala:186)
    [info]   at org.apache.spark.ml.tuning.CrossValidator$$anonfun$4.apply(CrossValidator.scala:164)
    [info]   at org.apache.spark.ml.tuning.CrossValidator$$anonfun$4.apply(CrossValidator.scala:144)
    [info]   at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
    [info]   at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
    [info]   at scala.collection.IndexedSeqOptimized$class.foreach(IndexedSeqOptimized.scala:33)
    [info]   at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:186)
    [info]   at scala.collection.TraversableLike$class.map(TraversableLike.scala:234)
    [info]   at scala.collection.mutable.ArrayOps$ofRef.map(ArrayOps.scala:186)
    [info]   at org.apache.spark.ml.tuning.CrossValidator.fit(CrossValidator.scala:144)

I was able to reproduce in dockerized testing on a debian-based image after building from source as well as using the published JAR

    resolvers += "GitHub Repo" at "https://raw.githubusercontent.com/CodingCat/xgboost/maven-repo/"

    libraryDependencies ++= Seq(
      "org.apache.spark" %% "spark-core" % "2.3.1",
      "org.apache.spark" %% "spark-mllib" % "2.3.1",
      "org.scalatest" %% "scalatest" % "3.0.3",
      "ml.dmlc" % "xgboost4j-spark" % "0.80"
    )

(as well as a few other spark/xgboost4j-spark release combinations). I wanted to see whether this was an artifact of my process or whether it was reflected in the main build tests as well.